### PR TITLE
Batchsize fix

### LINKED ISSUE #105: Apply batch_size of attack to classifier.
**Describe the bug**
Currently in every attack, there are several calls to classifier's predictor without specifying batch_size, i.e. using the default batch_size of the predictor (128). This leads to OOM error in some cases. So we need to pass batch_size of the attack to those call.

### PR DESCRIPTION
**Pull request by minhitbk**
Fixes # (issue)

Fixes #105 

## Type of change

Please check all relevant options.

- [X] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

- [ ] Test A
- [ ] Test B

**Test Configuration**:
- OS
- Python version
- ART version or commit number
- TensorFlow / Keras / PyTorch / MXNet version

# Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: art/attacks/hop_skip_jump.py [hunk 1]
```diff
@@ -34,7 +36,7 @@ class HopSkipJump(Attack):
     Paper link: https://arxiv.org/abs/1904.02144
     """
     attack_params = Attack.attack_params + ['targeted', 'norm', 'max_iter', 'max_eval',
-                                            'init_eval', 'init_size', 'curr_iter']
+                                            'init_eval', 'init_size', 'curr_iter', 'batch_size']
 
     def __init__(self, classifier, targeted=False, norm=2, max_iter=50, max_eval=10000, init_eval=100, init_size=100):
```
Comment (beat-buesser) [inline art/attacks/hop_skip_jump.py]:
```suggestion
    def __init__(self, classifier, targeted=False, norm=2, max_iter=50, max_eval=10000, init_eval=100, init_size=100, batch_size=1):
```

Could you also pleas update the parameter description with 
```
:param batch_size: Batch size
        :type batch_size: `int`
```

File: art/attacks/hop_skip_jump.py [hunk 2]
```diff
@@ -63,6 +65,7 @@ def __init__(self, classifier, targeted=False, norm=2, max_iter=50, max_eval=100
                   'init_eval': init_eval,
                   'init_size': init_size,
                   'curr_iter': 0,
+                  'batch_size': 1
```
Comment (beat-buesser) [inline art/attacks/hop_skip_jump.py]:
```suggestion
                  'batch_size': batch_size
```

File: art/attacks/boundary.py [hunk 3]
```diff
@@ -33,7 +35,7 @@ class BoundaryAttack(Attack):
     only requires final class prediction. Paper link: https://arxiv.org/abs/1712.04248
     """
     attack_params = Attack.attack_params + ['targeted', 'delta', 'epsilon', 'step_adapt', 'max_iter', 'num_trial',
-                                            'sample_size', 'init_size']
+                                            'sample_size', 'init_size', 'batch_size']
 
     def __init__(self, classifier, targeted=True, delta=0.01, epsilon=0.01, step_adapt=0.667, max_iter=5000,
                  num_trial=25, sample_size=20, init_size=100):
```
[the remark this answers is not in the document]
Comment (minhitbk) [reply art/attacks/boundary.py]: This attack currently does not support batching.

File: art/attacks/boundary.py [hunk 4]
```diff
@@ -68,6 +70,7 @@ def __init__(self, classifier, targeted=True, delta=0.01, epsilon=0.01, step_ada
                   'num_trial': num_trial,
                   'sample_size': sample_size,
                   'init_size': init_size,
+                  'batch_size': 1
```
[the remark this answers is not in the document]
Comment (minhitbk) [reply art/attacks/boundary.py]: This attack currently does not support batching.

File: art/attacks/boundary.py [hunk 5]
```diff
@@ -118,8 +121,11 @@ def generate(self, x, y=None, x_adv_init=None):
                 x_adv[ind] = self._perturb(x=val, y=-1, y_p=preds[ind], init_pred=init_preds[ind],
                                            adv_init=x_adv_init[ind], clip_min=clip_min, clip_max=clip_max)
 
+        if y is not None:
```
[the remark this answers is not in the document]
Comment (minhitbk) [reply art/attacks/boundary.py]: Because we use compute_success in the following line.

File: art/attacks/hop_skip_jump.py (hunk 1 above)
Comment (minhitbk) [reply art/attacks/hop_skip_jump.py]: This attack currently does not support batching.

File: art/attacks/hop_skip_jump.py (hunk 2 above)
Comment (minhitbk) [reply art/attacks/hop_skip_jump.py]: This attack currently does not support batching.

**Pull request merged**